osp/__init__.py

```diff
@@ -1 +1,60 @@
 __path__ = __import__('pkgutil').extend_path(__path__, __name__)
+
+# Patch RDFLib <= 5.0.0. See osp-core issue https://github.com/simphony/osp-core/issues/558 (the drive letter from the
+# path is stripped on Windows by the graph.Graph.serialize method of RDFLib <= 5.0.0).
+import sys as _osp_sys
+if _osp_sys.platform == 'win32':
+    import os as _osp_os
+    import ctypes as _osp_ctypes
+    import ctypes.wintypes as _osp_wintypes
+    from urllib.parse import urlparse as _osp_urlparse
+    import rdflib as _osp_rdflib
+
+
+    def _compare_version_leq(version, other_version):
+        """Compares two software version strings.
+
+        Receives two software version strings which are just numbers separated by dots and determines whether the first one
+        is less or equal than the second one.
+
+        Args:
+            version (str): first version string (number separated by dots).
+            other_version (str) : second version string (number separated by dots).
+
+        Returns:
+            bool: whether the first version string is less or equal than the second one.
+        """
+        version = map(int, version.split('.'))
+        other_version = map(int, other_version.split('.'))
+        for v, o in zip(version, other_version):
```

Comment:
I feel like this was too bulky and did not cover all the cases.
- You were comparing strings. Could lead to issues when comparing 2.10 with 2.2
- When everything was fine in the loop, you were only checking if the next number was larger than 0. Could lead to issues whem comparing 2.1 with 2.1.0.0.1.2.0.0 (probably not the biggest issue ;) )

Reply:
I wonder if we need a unittest for this...

Reply:
You are right, your solution is way less bulky and I did not consider the problem with the strings and the additional zeroes.

Reply:
I'll write a test for this.